shared/src/util/Numbers.ts

```diff
@@ -135,7 +135,9 @@ export function roundUpToNearestN(value: number, n: number): number {
 
 export function formatTokenAmount(amount: number, sigDigs = 4): string {
   //TODO: if we want to support more than one locale, we would need to add more logic here
-  if (amount > 1e13) {
+  if (amount > 1e34) {
+    return '∞';
```

Comment:
Hmm I'm not sure I like this. Can we make this more specific to the problem you're trying to solve?